lib/jellyfish/room_service.ex

```diff
@@ -55,27 +55,39 @@ defmodule Jellyfish.RoomService do
     |> Enum.reject(&(&1 == nil))
   end
 
-  @spec create_room(Room.max_peers(), String.t()) ::
+  @spec create_room(Room.max_peers(), String.t(), Room.id() | nil) ::
           {:ok, Room.t(), String.t()} | {:error, :invalid_max_peers | :invalid_video_codec}
-  def create_room(max_peers, video_codec) do
-    {node_resources, failed_nodes} =
-      :rpc.multicall(Jellyfish.RoomService, :get_resource_usage, [])
-
-    if Enum.count(failed_nodes) > 0 do
-      Logger.warn(
-        "Couldn't get resource usage of the following nodes. Reason: nodes don't exist. Nodes: #{inspect(failed_nodes)}"
-      )
-    end
 
-    {min_node, _room_size} =
-      Enum.min_by(node_resources, fn {_node_name, room_num} -> room_num end)
+  # NOTICE:
+  # #########
+  # for purpose of testing linking rooms between jellyfishes in cluster
+  # we need to manually handle which client we want to create room on.
+  # So a code below, that creates room on node chosen by resource usage is temorary commented
+  # #########
 
-    if Enum.count(node_resources) > 1 do
-      Logger.info("Node with least used resources is #{inspect(min_node)}")
-      GenServer.call({__MODULE__, min_node}, {:create_room, max_peers, video_codec})
-    else
-      GenServer.call(__MODULE__, {:create_room, max_peers, video_codec})
-    end
+  # def create_room(max_peers, video_codec, id \\ nil) do
+  #   {node_resources, failed_nodes} =
+  #     :rpc.multicall(Jellyfish.RoomService, :get_resource_usage, [])
+
+  #   if Enum.count(failed_nodes) > 0 do
+  #     Logger.warn(
+  #       "Couldn't get resource usage of the following nodes. Reason: nodes don't exist. Nodes: #{inspect(failed_nodes)}"
+  #     )
+  #   end
+
+  #   {min_node, _room_size} =
+  #     Enum.min_by(node_resources, fn {_node_name, room_num} -> room_num end)
+
+  #   if Enum.count(node_resources) > 1 do
+  #     Logger.info("Node with least used resources is #{inspect(min_node)}")
+  #     GenServer.call({__MODULE__, min_node}, {:create_room, max_peers, video_codec, id})
+  #   else
+  #     GenServer.call(__MODULE__, {:create_room, max_peers, video_codec, id})
+  #   end
+  # end
```

Comment:
It can be very easily solved if room_id is different than nil we start It on a current node in all other situations we use load balancing.

Reply:
That was my first idea but IMO that's not a good assumption. API should be clear and consequent. Passing or not room id value should not change the logic of how  and where room is created. We can't leave such a surprise in API. If we want to do it we should add a flag like `distributed: true` to make it explicit.

Reply:
From one point it's true but, from the second point of view passing ID when you create room on the first jellyfish is useless. When you pass ID and want to use load balancing there is a question of how it should be handled when some room with this ID already exists etc.